### PR TITLE
chore: initial example and walk-through

### PR DESCRIPTION
**Pull request by mariux**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,71 @@
...
+
+- `environments/prod/vpc`
+- `environments/prod/ec2-instance`
+- `environments/staging/vpc`
+- `environments/staging/ec2-instance`
+
+# world with terramate
+
+- rename terramate config to disable production safeguards for this demo
+  - `cp terramate.tm.hcl.example terramate.tm.hcl`
```
Comment (kkmlr) [inline README.md:13]: terramate.tm.hcl already exists, so either remove the terramate.tm.hcl.example file, or remove this step

File: README.md [hunk 2]
```diff
@@ -1,2 +1,71 @@
...
+  - `terramate create --all-terraform`
+  - This will create a `stacks.tm.hcl` in each directory giving each stack a name and an unique ID (UUID)
+- validate stacks are detected by listing them
+  - `terramate list`
+- commit the new found configuration for Terramate Stacks
+  - `git add -A && git commit -m 'feat: initialize Terramate stacks'`
+- initialize terraform in all stacks an validate the configuration
+  - `terramate run -- terraform init`
+  - `terramate run -- terraform validate`
+  - The Terramate configuration in `terramate.tm.hcl` takes care that we cache the providers in a `.tf_plugin_cache_dir` by setting the Terraform environment variable `TF_PLUGIN_CACHE_DIR`. This speeds up `terraform init` phase as each provider version is downloaded only once.
```
Comment (kkmlr) [inline README.md]:
```suggestion
  - The Terramate configuration in `terramate.tm.hcl` ensures that we cache the providers in a `.tf_plugin_cache_dir` by setting the Terraform environment variable `TF_PLUGIN_CACHE_DIR`. This speeds up `terraform init` phase as each provider version is downloaded only once.
```

that's actually a pretty neat trick btw

File: README.md [hunk 3]
```diff
@@ -1,2 +1,71 @@
...
+
+- `generate_backend.tm.hcl.example` rename to enable backend generation
+- run `terramate generate` to generate `gen_backend.tf` in each directory
+- run `terramate run -- rm backend.tf` to remove old config
+
+### generate providers
+
+- `generate_providers.tm.hcl.example` rename to enable providers generation
+- run `terramate generate` to generate `gen_providers.tf` in each directory
+- run `terramate run -- rm providers.tf` to remove old config
```
Comment (kkmlr) [inline README.md:54]: image

you miss spelled providers in the file names

File: README.md [hunk 4]
```diff
@@ -1,2 +1,71 @@
...
+- set global `tf_provider_null_version_constraint` on any level e.g. in stack or per environment in a e.g. `config.tm.hcl` file:
+  ```hcl
+  globals {
+    tf_provider_null_version_constraint = "~> 3.0"
+  }
+  ```
```
Comment (kkmlr) [inline README.md]:
```suggestion
- set global `tf_provider_null_version_constraint` on any level e.g. in stack or per environment in a e.g. `/environments/staging/config.tm.hcl` file:
  ```hcl
  globals {
    tf_provider_null_version_constraint = "3.0"
  }
  ```
    
```

Comment (kkmlr) [inline README.md]: and add 
```
- run `terramate generate` to see the changes
```
(I can't do this in the suggestion as it breaks github's markdown)

Comment (kkmlr) [reply README.md]: tf_provider_null_version_constraint is already set to `~> 3.0` at the top level. It's clear what you mean by this, but worth being a little more explicit and showing an actual change that can be demonstrated
however, changing the provider here will require a re-init, so perhaps best not to

**Review by kkmlr — COMMENTED** [review]
A few comments but apart from s/povider/provider it seems fine

File: README.md (hunk 1 above)
Comment (mariux) [reply README.md:13]: removed

**Review by soerenmartius — APPROVED** [review]
lgtm

**Pull request merged**